mongodbatlas/resource_mongodbatlas_cluster_test.go

```diff
@@ -675,6 +675,101 @@ func TestAccClusterRSCluster_AWSWithLabels(t *testing.T) {
 	})
 }
 
+func TestAccClusterRSCluster_WithTags(t *testing.T) {
+	var (
+		cluster                matlas.Cluster
+		resourceName           = "mongodbatlas_cluster.test"
+		dataSourceName         = "data.mongodbatlas_cluster.test"
+		dataSourceClustersName = "data.mongodbatlas_clusters.test"
+		orgID                  = os.Getenv("MONGODB_ATLAS_ORG_ID")
+		projectName            = acctest.RandomWithPrefix("test-acc")
+		name                   = fmt.Sprintf("testAcc-%s-%s-%s", "AWS", "M10", acctest.RandString(1))
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheckBasic(t) },
+		ProtoV6ProviderFactories: testAccProviderV6Factories,
+		CheckDestroy:             testAccCheckMongoDBAtlasClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMongoDBAtlasClusterConfigWithTags(orgID, projectName, name, "false", "M10", "EU_CENTRAL_1", []matlas.Tag{}),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMongoDBAtlasClusterExists(resourceName, &cluster),
+					testAccCheckMongoDBAtlasClusterAttributes(&cluster, name),
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					resource.TestCheckResourceAttrSet(resourceName, "mongo_uri"),
+					resource.TestCheckResourceAttr(resourceName, "tags.#", "0"),
+					resource.TestCheckResourceAttr(dataSourceName, "tags.#", "0"),
+					resource.TestCheckResourceAttr(dataSourceClustersName, "results.0.tags.#", "0"),
+				),
+			},
+			{
+				Config: testAccMongoDBAtlasClusterConfigWithTags(orgID, projectName, name, "false", "M10", "EU_CENTRAL_1",
+					[]matlas.Tag{
+						{
+							Key:   "key 1",
+							Value: "value 1",
+						},
+						{
+							Key:   "key 2",
+							Value: "value 2",
+						},
+					},
+				),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMongoDBAtlasClusterExists(resourceName, &cluster),
+					testAccCheckMongoDBAtlasClusterAttributes(&cluster, name),
+					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					resource.TestCheckResourceAttrSet(resourceName, "mongo_uri"),
+					resource.TestCheckResourceAttr(resourceName, "tags.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "tags.0.key", "key 1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.0.value", "value 1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.1.key", "key 2"),
+					resource.TestCheckResourceAttr(resourceName, "tags.1.value", "value 2"),
+					resource.TestCheckResourceAttr(dataSourceName, "tags.#", "2"),
+					resource.TestCheckResourceAttr(dataSourceName, "tags.0.key", "key 1"),
+					resource.TestCheckResourceAttr(dataSourceName, "tags.0.value", "value 1"),
+					resource.TestCheckResourceAttr(dataSourceName, "tags.1.key", "key 2"),
```

Comment:
it'd be nice to have a less verbose way to check tags, or other attributes

Reply:
Agree.
I have looked into one alternative to simplify defining assertions of the terraform state, this is the PoC: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1448.